src/System.Windows.Forms/src/System/Windows/Forms/Printing/PrintPreviewControl.PrintPreviewControlAccessibleObject.cs

```diff
@@ -78,5 +88,124 @@ internal override UiaCore.IRawElementProviderFragmentRoot FragmentRoot
+                        break;
+                    case ScrollAmount.ScrollAmount_SmallDecrement:
+                        scrollValue = owner._vScrollBar.Value - owner._vScrollBar.SmallChange;
+                        owner._vScrollBar.Value = (scrollValue < owner._vScrollBar.Minimum ? owner._vScrollBar.Minimum : scrollValue);
+                        break;
+                }
+            }
+
+            return HRESULT.S_OK;
+        }
+
+        HRESULT IScrollProvider.Interface.SetScrollPercent(double horizontalPercent, double verticalPercent)
+        {
+            if (!this.TryGetOwnerAs(out PrintPreviewControl? owner))
+            {
+                return HRESULT.S_OK;
+            }
+
+            int scrollValue = 0;
+            if (owner._hScrollBar.Visible && verticalPercent >= 0 && verticalPercent <= 100)
+            {
+                scrollValue = owner._hScrollBar.Minimum + (int)((owner._hScrollBar.Maximum - owner._hScrollBar.Minimum) * verticalPercent);
+                owner._hScrollBar.Value = scrollValue;
+            }
+            else if (owner._vScrollBar.Visible && verticalPercent >= 0 && verticalPercent <= 100)
+            {
+                scrollValue = owner._vScrollBar.Minimum + (int)((owner._vScrollBar.Maximum - owner._vScrollBar.Minimum) * verticalPercent);
+                owner._vScrollBar.Value = scrollValue;
+            }
+
+            return HRESULT.S_OK;
+        }
+
+        public double HorizontalScrollPercent
+        {
+            get
+            {
+                if (this.TryGetOwnerAs(out PrintPreviewControl? owner) && owner._hScrollBar.Visible)
+                {
+                    double percent = owner._hScrollBar.Value * 100.0 / (owner._hScrollBar.Maximum - owner._hScrollBar.LargeChange);
+                    return percent > 100 ? 100 : percent;
+                }
+
+                return 0;
+            }
+        }
+
+        public double VerticalScrollPercent
+        {
+            get
+            {
+                if (this.TryGetOwnerAs(out PrintPreviewControl? owner) && owner._vScrollBar.Visible)
+                {
+                    double percent = owner._vScrollBar.Value * 100.0 / (owner._vScrollBar.Maximum - owner._vScrollBar.LargeChange);
+                    return percent > 100 ? 100 : percent;
+                }
+
+                return 0;
+            }
+        }
+
+        public double HorizontalViewSize => this.TryGetOwnerAs(out PrintPreviewControl? owner)
+            && owner._hScrollBar.Visible ? owner.HorizontalViewSize : 100;
+
+        public double VerticalViewSize => this.TryGetOwnerAs(out PrintPreviewControl? owner)
+            && owner._vScrollBar.Visible ? owner.VerticalViewSize : 100;
+
+        BOOL IScrollProvider.Interface.HorizontallyScrollable =>
+            this.TryGetOwnerAs(out PrintPreviewControl? owner) ? (BOOL)owner._hScrollBar.Visible : (BOOL)false;
```

Comment:
nit: I don't think the casting is necessary here.
```suggestion
            this.TryGetOwnerAs(out PrintPreviewControl? owner) ? owner._hScrollBar.Visible : false;
```